### PR TITLE
Get public IP of a device and send it to CoolMenu

### PR DESCRIPTION
**Pull request by LeSuedois**
like said above, get the public IP of a device for geolocation. 
sends the IP each time the connect() function is called to ensure reliable service.

### CODE REVIEW

File: src/CoolWifi.cpp [hunk 1]
```diff
@@ -628,3 +628,47 @@ bool CoolWifi::addWifi(String ssid, String pass) {
...
+/**
+ *  CoolWifi::getExternalIP():
+ *  This method is provided to print the
+ *  public IP of a device to a String
+ */
+String CoolWifi::getExternalIP()
+{
+#if DEBUG == 1
+
+  Serial.println("Entering CoolWifi.addWifi() ");
```
Comment (3rwww1) [inline src/CoolWifi.cpp]: Log statement does not match the function name... Copy paste ?

File: src/CoolWifi.cpp [hunk 2]
```diff
@@ -628,3 +628,47 @@ bool CoolWifi::addWifi(String ssid, String pass) {
...
+
+#endif
+
+  WiFiClient client;
+  String IP;
+  if (!client.connect("api.ipify.org", 80)) {
+    Serial.println(F ("Failed to connect with 'api.ipify.org' !"));
+  }
```
Comment (3rwww1) [inline src/CoolWifi.cpp]: per LLVM specifications (and almost all others), braces for `else` statement should look like 
```
} else {
```
not
```
}
else {
```

File: src/CoolWifi.cpp [hunk 3]
```diff
@@ -628,3 +628,47 @@ bool CoolWifi::addWifi(String ssid, String pass) {
...
+  else {
+  int timeout = millis() + 800;
```
Comment (3rwww1) [inline src/CoolWifi.cpp]: `int timeout` should be indented 2 spaces to the right

File: src/CoolWifi.cpp [hunk 4]
```diff
@@ -628,3 +628,47 @@ bool CoolWifi::addWifi(String ssid, String pass) {
...
+    client.print("GET /?format=json HTTP/1.1\r\nHost: api.ipify.org\r\n\r\n");
```
Comment (3rwww1) [inline src/CoolWifi.cpp]: You don't need to use the `?format=json` here. So you can avoid parsing the JSON later

File: src/CoolWifi.cpp [hunk 5]
```diff
@@ -628,3 +628,47 @@ bool CoolWifi::addWifi(String ssid, String pass) {
...
+    client.stop();
+
+#if DEBUG == 1
+    Serial.print("Received Message : ");
+    Serial.println(IP);
+#endif
+    
+  }
+  //return only the IP in the string
+  return IP.substring(IP.indexOf("{")+6,IP.lastIndexOf("}"));
```
Comment (3rwww1) [inline src/CoolWifi.cpp:672]: same comment as above. see ipify.org homepage, navigate to section "API Usage" and see the first example :

| API URI                   | Response Type | Sample Output (IPv4) |
| --- | --- | --- |
| https://api.ipify.org | text                  | 98.207.254.136          |

in short, you don't need JSON at all :)

**Review by 3rwww1 — CHANGES_REQUESTED** [review]
Looks good to me, but require some minor changes in my humble opinion

File: src/CoolWifi.cpp (hunk 4 above)
Comment (LeSuedois) [reply src/CoolWifi.cpp]: I tried and it doesn't worked the "GET /", I just got Timeout... that's also the reason why I count letters below.

File: src/CoolWifi.cpp (hunk 5 above)
Comment (LeSuedois) [reply src/CoolWifi.cpp:672]: ok, I retried and now it works, no Idea why it didn't worked two days ago..
But it's easier to get a correct IP from a good old JSON due to the variable length and the fact that ipify doesn't send just the IP but more something like that : 
'''
1.1 200 OK
Server: Cowboy
Connection: keep-alive
Content-Type: text/plain
Vary: Origin
Date: Tue, 20 Feb 2018 08:10:08 GMT
Content-Length: 13
Via: 1.1 vegur

178.20.51.116
'''
Of course I could get the IP by selecting from '\r\n\r\n' to the end but what does that change? 
please comment

**Review by 3rwww1 — APPROVED** [review]
Still need to try real HTTP Client for error handling

**Pull request merged**